src/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp

```diff
@@ -858,9 +866,17 @@ std::shared_ptr<HttpResponse> CurlHttpClient::MakeRequest(const std::shared_ptr<
             response->SetClientErrorType(CoreErrors::NETWORK_CONNECTION);
             Aws::StringStream ss;
             ss << "curlCode: " << curlResponseCode << ", " << curl_easy_strerror(curlResponseCode);
+            if (errorBuffer.GetUnderlyingData() && errorBuffer.GetSize() >= CURL_ERROR_SIZE) {
+              ss << "; Details: ";
+              for(size_t i = 0; i < CURL_ERROR_SIZE; ++i) {
```

Comment:
any reason we have to loop over the error buffer instead of just streaming? the following works as expected

```c++
#include <aws/core/Aws.h>

using namespace Aws;
using namespace Aws::Utils;

auto main() -> int {
  SDKOptions options{};
  options.loggingOptions.logLevel = Logging::LogLevel::Debug;
  InitAPI(options);
  {
    ByteBuffer buffer{"some error string"};
    std::cout << buffer.GetUnderlyingData() << "\n";
  }
  ShutdownAPI(options);
  return 0;
}
```

Reply:
https://curl.se/libcurl/c/CURLOPT_ERRORBUFFER.html

It is just a defensive programming against libcurl not nulling the buffer.
Well I'm already setting the null terminator at the beginning, probably it could replace the loop.